backend/cmd/api/main.go

```diff
@@ -44,11 +45,30 @@ func playgroundHandler() gin.HandlerFunc {
 	}
 }
 
+func authenticateRequestWithXApiKey(apiKey string, c *gin.Context) error {
+	a := c.Request.Header.Get("X-API-Key")
+	if a == apiKey {
+		return nil
+	}
+
+	var err error
+	if a == "" {
+		err = errors.New("missing x-api-key header")
+	} else {
+		err = errors.New("failed to authenticate")
+	}
+
+	c.Status(401)
+	_, _ = c.Writer.WriteString("Unauthorized")
```

Comment:
```suggestion
	c.String("Unauthorized")
	c.AbortWithStatus(http.StatusUnauthorized)
```

Gin provides many nice functions like for example https://pkg.go.dev/github.com/gin-gonic/gin#Context.AbortWithStatusJSON